File: catalog/backends/python-flask/patterns/audit_log_middleware.py
import logging
from flask import request

# Pattern: SIEM — structured audit log for auth-relevant events
# Adapt: log fields relevant to your domain's security events
audit = logging.getLogger("audit")


def audit_log_after_request(response):
    audit.info(
        "request",
        extra={
            "method": request.method,
            "path": request.path,
            "status": response.status_code,
            "remote": request.remote_addr,
        },
    )
    return response

File: catalog/backends/php-laravel/patterns/AuditLogMiddleware.php
<?php

namespace App\Http\Middleware;

use Closure;
use Illuminate\Http\Request;
use Illuminate\Support\Facades\Log;

// Pattern: SIEM — structured audit log for auth-relevant events
// Adapt: log fields relevant to your domain's security events

class AuditLogMiddleware
{
    public function handle(Request $request, Closure $next)
    {
        $response = $next($request);
        Log::channel('stdout')->info('request', [
            'method' => $request->method(),
            'path' => $request->path(),
            'status' => $response->getStatusCode(),
            'remote' => $request->ip(),
        ]);
        return $response;
    }
}
